packages/hawtio/src/preferences/HomePreferences.tsx

```diff
@@ -1,13 +1,46 @@
 import { Button, CardBody, Form, FormGroup, FormSection, Switch } from '@patternfly/react-core'
 import React, { useState } from 'react'
-import { log } from './globals'
+
+type LocalStorageFieldData<T> = {
+  localStorageKey: string
+  defaultValue: T
+  updateFunction: React.Dispatch<React.SetStateAction<T>>
+}
 
 export const HomePreferences: React.FunctionComponent = () => {
-  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(true)
+  const LOCAL_STORAGE_VERTICAL_NAVIGATION = 'local_storage_vertical_navigation'
+  const DEFAULT_VALUE_VERTICAL_NAVIGATION = true
+
+  const VERTICAL_NAVIGATION_INITIAL_VALUE: boolean =
+    localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) !== null
+      ? localStorage.getItem(LOCAL_STORAGE_VERTICAL_NAVIGATION) === 'true'
+      : DEFAULT_VALUE_VERTICAL_NAVIGATION
+
+  const [defaultVerticalNavState, setDefaultVerticalNavState] = useState(VERTICAL_NAVIGATION_INITIAL_VALUE)
+
+  const VERTICAL_NAVIGATION_FIELD_PARAMETERS: LocalStorageFieldData<boolean> = {
+    localStorageKey: LOCAL_STORAGE_VERTICAL_NAVIGATION,
+    defaultValue: true,
+    updateFunction: setDefaultVerticalNavState,
+  }
+
+  const FIELDS_TO_RESET = [VERTICAL_NAVIGATION_FIELD_PARAMETERS]
 
   const reset = () => {
-    // TODO: impl
-    log.info('TODO - Reset settings')
+    FIELDS_TO_RESET.forEach(field => {
+      localStorage.removeItem(field.localStorageKey)
+      field.updateFunction(field.defaultValue)
+    })
+  }
+
+  const savingToLocalStorage = <T,>(
```

Comment:
I don't think we need to do this in such a complex way.

Reply:
Clear would remove all the settings, moreover, everything on the localstorage, not just the one on that page of the settings, so its generally recommended to avoid it.

Ill check it tomorrow anyway and show a gif too if it is as I think